src/cpp/src/utils.cpp

```diff
@@ -443,17 +443,7 @@ compile_decoder_for_npu(const std::shared_ptr<ov::Model>& model,
         kv_desc.max_prompt_len = pop_int_and_cast(properties, "MAX_PROMPT_LEN").value_or(1024u);
         kv_desc.min_response_len = pop_int_and_cast(properties, "MIN_RESPONSE_LEN").value_or(128u);
         update_npu_config(properties, model, kv_pos, kv_desc);
-        auto cache_mode = get_option<CacheMode>(config, ov::cache_mode.name());
-        // NB: Select OPTIMIZE_SPEED with model_path isn't provided
-        if ((cache_mode.has_value() && *cache_mode == CacheMode::OPTIMIZE_SPEED)) {
-            compiled = ov::genai::utils::singleton_core().compile_model(model, "NPU", properties);
-        } else if (model_path.empty()) {
-            // Set config to OPTIMIZE_SPEED
-            properties[ov::cache_mode.name()] = CacheMode::OPTIMIZE_SPEED;
-            compiled = ov::genai::utils::singleton_core().compile_model(model, "NPU", properties);
-        } else {
-            compiled = ov::genai::utils::singleton_core().compile_model(model_path, "NPU", properties);
```

Comment:
could you please also drop `model_path` from places, where it was introduced in https://github.com/openvinotoolkit/openvino.genai/pull/1677/files#diff-b051a40fb8e3f841d2dbe04cea08b010afc0aad852d12d8bd3a0267c70a32a6eR43